Sources/ComposableArchitecture/Effects/Throttling.swift

```diff
@@ -45,7 +47,12 @@ extension Effect {
             scheduler: scheduler
           )
           .handleEvents(
-            receiveOutput: { _ in throttleLock.sync { throttleTimes[id] = scheduler.now } }
+            receiveOutput: { _ in
+                throttleLock.sync {
+                    throttleTimes[id] = scheduler.now
+                    throttleValues[id] = nil
```

Comment:
This fixes the bug (when using a strategy different than `latest`), while ensuring we clean up shared state